core/domain/exp_domain.py

```diff
@@ -3138,6 +3186,61 @@ def _convert_states_v51_dict_to_v52_dict(
 
         return states_dict
 
+    @classmethod
+    def _convert_states_v52_dict_to_v53_dict(
+        cls, states_dict: Dict[str, state_domain.StateDict]
+    ) -> Dict[str, state_domain.StateDict]:
+        """Converts from version 52 to 53. Version 53 makes sure the links used
+        in the non-interactive link component are all 'https' links.
+
+        Args:
+            states_dict: dict. A dict where each key-value pair represents,
+                respectively, a state name and a dict used to initialize a
+                State domain object.
+
+        Returns:
+            dict. The converted states_dict.
+        """
+
+        for state_dict in states_dict.values():
+            valid_html = (
+                cls.fix_non_interactive_links(state_dict['content']['html']))
+
+            state_dict['content']['html'] = valid_html
+
+        for state_dict in states_dict.values():
+            translations_mapping = (
+                state_dict['written_translations']['translations_mapping'])
+            new_translations_mapping = {}
+            for content_id, translation_item in translations_mapping.items():
+                if content_id != 'content':
+                    new_translations_mapping[content_id] = translation_item
```

Comment:
Why is this needed if we do the same thing at line 3237

Reply:
This line is redundant, removed.